src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,66 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String COMMA = ",";
+    private static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        List<String> dataFromFile = readFromFile(fromFileName);
+        int supplyAmount = calculateTotal(dataFromFile, SUPPLY);
+        int buyAmount = calculateTotal(dataFromFile, BUY);
+        int result = supplyAmount - buyAmount;
+        List<String> report = createReport(supplyAmount, buyAmount, result);
+        writeToFile(toFileName, report);
+    }
+
+    private List<String> readFromFile(String fileName) {
+        try (BufferedReader reader = Files.newBufferedReader(Paths.get(fileName))) {
+            return Files.readAllLines(Paths.get(fileName));
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read the file " + fileName, e);
+        }
+    }
+
+    private int calculateTotal(List<String> data, String operationType) {
+        int total = 0;
+        for (String line : data) {
+            String[] eachElement = line.split(",");
+            if (eachElement[0].equals(operationType)) {
+                total += Integer.parseInt(eachElement[1]);
+            }
+        }
+        return total;
+    }
+
+    private List<String> createReport(int supplyAmount, int buyAmount, int result) {
+        StringBuilder firstLine = new StringBuilder();
```

Comment:
Why you need 3 StringBuilder objects? 
You can create one and return toString, instead of List<String>